### PR TITLE
Blog build

### PR DESCRIPTION
**Pull request by apgmartin**

### CODE REVIEW

File: article.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    <title>Article Page</title>
+    <link rel="stylesheet" type="text/css" href="css/global.css">
+    <link rel="stylesheet" type="text/css" href="css/article.css">
+    <link rel="icon" type="image" href="images/amlogo.jpg">
+</head>
+<body>
+<header class="<?php echo $class; ?> header_image">
+    <div id="blackbar"></div>
+    <a href="blog.php" id="homebutton" class="navbutton">Back To Blog Main</a>
+    <a href="home.html" id="latestblog" class="navbutton">Previous Post</a>
```
Comment (rmjjjt) [inline article.php]: This takes you back to the home page. Previous post?

File: article.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    <?php
+    echo '<div id="titleblock">
+        <h1>' . $post->getTitle() . '</h1>
+        <h4>' . $post->getDate() . '</h4>
+    </div>
+</header>
+<div id="content_container">
+    <p>' . $post->getText() . '</p>
+    <a href="blog.php" class="bottom_links" id="blog_page_bottom_link">List of Blogs</a>
+    <a href="http://google.co.uk" id="prev_post_bottom_link" class="bottom_links">Previous Post</a>
```
Comment (rmjjjt) [inline article.php]: Google!

File: blog.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    <title>My Blog</title>
+    <link rel="stylesheet" type="text/css" href="css/global.css">
+    <link rel="stylesheet" type="text/css" href="css/blog.css">
+    <link rel="icon" type="image" href="images/amlogo.jpg">
+</head>
+<body>
+<header>
+    <div id="blackbar"></div>
+    <a href="home.html" id="homebutton" class="navbutton">Back To Homepage</a>
+    <a href="index.php" id="latestblog" class="navbutton">Latest Blog Post</a>
```
Comment (rmjjjt) [inline blog.php]: This takes you to the splash page - going to the latest post article would make more sense - that's what the button says!

File: blog.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    <div id="headercontent">
+        <img src="images/alipic2.jpg" id="profilepic" alt="Ali Martin">
+        <h1 id="blogname">Ali Martin, Software Dev</h1>
+        <p id="blogtagline">Follow my journey as a trainee software developer.</p>
+        <div id="socialmediacontainer">
+            <a href="https://github.com/apgmartin" target="_blank"><img src="images/github.jpg" class="socialmedia" alt="GitHub"></a>
```
Comment (rmjjjt) [inline blog.php]: This image doesn't make the others - white background. The others all have transparent backgrounds. Continuity (MPO, not me!).

File: css/article.css [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+#titleblock {
+    text-align: center;
+    padding-top: 90px;
+}
+
+#titleblock h1 {
+    font-size: 60px;
+    margin-bottom: 10px;
+    margin-top: 0;
+    color: white;
```
Comment (rmjjjt) [inline css/article.css]: Duplicate of styling - color, stroke, shadow replicated - could be one class.

File: article.php (hunk 1 above)
Comment (apgmartin) [inline article.php]: Corrected

File: article.php (hunk 2 above)
Comment (apgmartin) [inline article.php]: Corrected

File: blog.php (hunk 4 above)
Comment (apgmartin) [inline blog.php]: Corrected.

File: css/article.css (hunk 5 above)
Comment (apgmartin) [inline css/article.css]: corrected.

File: blog.php (hunk 3 above)
Comment (apgmartin) [inline blog.php]: Corrected.

File: article.php [hunk 6]
```diff
@@ -0,0 +1,75 @@
+<?php
+require('create_blogpost.php');
```
Comment (aimee-wyld) [inline article.php]: require doesn't need brackets - apparently it's a statement rather than a function?

File: article.php [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+require ('random_class.php');
```
Comment (aimee-wyld) [inline article.php]: require doesn't need brackets

File: blog.php [hunk 8]
```diff
@@ -0,0 +1,53 @@
+<?php
+require('create_blogpost.php');
```
Comment (aimee-wyld) [inline blog.php]: require doesn't need brackets

File: create_blogpost.php [hunk 9]
```diff
@@ -0,0 +1,13 @@
+<?php
+require('blog_class.php');
```
Comment (aimee-wyld) [inline create_blogpost.php]: brackets

File: css/article.css [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+
+header {
+    border-bottom: 10px solid #A2D729;
+    height: 420px;
+    z-index: 2;
+}
+
+.header_image {
+    background-size: cover;
+    background-color: white;
```
Comment (aimee-wyld) [inline css/article.css]: RGB/hex?

File: css/article.css [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+.rand_class2 {
+    background-image: url("../images/mountains.jpeg");
+    background-position: 50% 84%;
+}
+
+.rand_class3 {
+    background-image: url("../images/forest.jpeg");
+    background-position: 50% 0%;
```
Comment (aimee-wyld) [inline css/article.css]: do you need the % after 0?

File: css/article.css [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+.nav_button {
+    position: fixed;
+    top: 0;
+    font-size: 18px;
+    font-weight: bold;
+    text-align: center;
+    width: 120px;
+    color: black;
```
Comment (aimee-wyld) [inline css/article.css]: rgb/hex?

File: css/article.css [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+    padding: 10px 10px 7px 10px;
+    border: 5px solid #FF6700;
+    text-decoration: none;
+    transition: 0.5s;
+}
+
+.nav_button:hover {
+    border: 5px solid #A2D729;
+    background-color: rgba(0,0,0,0.7);
+    color: white;
```
Comment (aimee-wyld) [inline css/article.css]: rgb/hex?

File: css/article.css [hunk 14]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+#social_media_links {
+    width: 100%;
+    height: 65px;
+    padding: 5px;
+    background-color: rgba(255,255,255,0.8);
+    overflow: hidden;
+    text-align: center;
+    margin: 50px 0px 0px 0px;
```
Comment (aimee-wyld) [inline css/article.css]: don't need px after 0?

File: css/article.css [hunk 15]
```diff
@@ -0,0 +1,204 @@
...
+    z-index: -5;
+    overflow: auto;
+}
+
+#blog_content p {
+    margin-top: 40px;
+    font-size: 14px;
+    text-align: left;
+    margin-bottom: 40px;
+    color: white;
```
Comment (aimee-wyld) [inline css/article.css]: rgb/hex?

File: css/article.css [hunk 16]
```diff
@@ -0,0 +1,204 @@
...
+.footer_links {
+    background-color: rgb(0, 0, 0);
+    color: rgb(255, 255, 255);
+    padding: 8px;
+    -moz-border-radius: 5px;
+    -webkit-border-radius: 5px;
+    border-radius: 5px;
+    font-size: 15px;
+    transition: 1s;
+    border: 4px solid black;
```
Comment (aimee-wyld) [inline css/article.css]: rgb/hex?

File: css/article.css [hunk 17]
```diff
@@ -0,0 +1,204 @@
...
+    width: 140px;
+}
+
+.footer_links:hover {
+    background-color: white;
```
Comment (aimee-wyld) [inline css/article.css]: rgb/hex?

File: css/article.css [hunk 18]
```diff
@@ -0,0 +1,204 @@
...
+    color: black;
```
Comment (aimee-wyld) [inline css/article.css]: rgb/hex?

File: css/blog.css [hunk 19]
```diff
@@ -0,0 +1,184 @@
...
+    width: 100%;
+    min-width: 600px;
+    background-color: #0B2F37;
+}
+
+header {
+    border-bottom: 10px solid #A2D729;
+    background-image: url("../images/ocean.jpg");
+    background-size: cover;
+    background-position: 50% 0%;
```
Comment (aimee-wyld) [inline css/blog.css]: do you need % for 0?

File: css/blog.css [hunk 20]
```diff
@@ -0,0 +1,184 @@
...
+    z-index: 1;
+}
+
+.nav_button {
+    position: fixed;
+    font-size: 18px;
+    font-weight: bold;
+    text-align: center;
+    width: 120px;
+    color: black;
```
Comment (aimee-wyld) [inline css/blog.css]: rgb/hex?

File: css/blog.css [hunk 21]
```diff
@@ -0,0 +1,184 @@
...
+    padding: 10px 10px 7px 10px;
+    border: 5px solid #FF6700;
+    text-decoration: none;
+    transition: 0.5s;
+}
+
+.nav_button:hover {
+    border: 5px solid #A2D729;
+    background-color: rgba(0,0,0,0.7);
+    color: white;
```
Comment (aimee-wyld) [inline css/blog.css]: rgb/hex?

File: css/blog.css [hunk 22]
```diff
@@ -0,0 +1,184 @@
...
+    margin: auto;
+    text-align: center;
+}
+
+#profile_pic {
+    height: 150px;
+    -moz-border-radius: 50%;
+    -webkit-border-radius: 50%;
+    border-radius: 50%;
+    margin: 30px 0px 5px 0px;
```
Comment (aimee-wyld) [inline css/blog.css]: don't need px for 0?

File: css/blog.css [hunk 23]
```diff
@@ -0,0 +1,184 @@
...
+    padding: 20px 30px 20px 30px;
+}
+
+.author_pic {
+    width: 20%;
+    float: left;
+    -moz-border-radius: 50%;
+    -webkit-border-radius: 50%;
+    border-radius: 50%;
+    border: 2px solid black;
```
Comment (aimee-wyld) [inline css/blog.css]: rgb/hex?

File: css/index.css [hunk 24]
```diff
@@ -71,12 +71,66 @@ a:hover .pc {
...
+.blog_post h4 {
+    font-size: 12px;
+    color: white;
+    text-align: right;
+    transition: 0.5s;
+}
+
+.blog_post p {
+    font-size: 14px;
+    color: white;
```
Comment (aimee-wyld) [inline css/index.css]: rgb/hex?

File: index.php [hunk 25]
```diff
@@ -0,0 +1,41 @@
+<?php
+require ('latest_blogs.php');
```
Comment (aimee-wyld) [inline index.php]: brackets

File: latest_blogs.php [hunk 26]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+require('create_blogpost.php');
```
Comment (aimee-wyld) [inline latest_blogs.php]: brackets

File: tests/blog_class.php [hunk 27]
```diff
@@ -0,0 +1,19 @@
+<?php
+require ('../blog_class.php');
```
Comment (aimee-wyld) [inline tests/blog_class.php]: brackets

File: tests/random_class.php [hunk 28]
```diff
@@ -0,0 +1,18 @@
+<?php
+require ('../random_class.php');
```
Comment (aimee-wyld) [inline tests/random_class.php]: brackets

File: article.php (hunk 6 above)
Comment (apgmartin) [inline article.php]: done

File: article.php (hunk 7 above)
Comment (apgmartin) [inline article.php]: done

File: blog.php (hunk 8 above)
Comment (apgmartin) [inline blog.php]: done

File: create_blogpost.php (hunk 9 above)
Comment (apgmartin) [inline create_blogpost.php]: done

File: css/article.css (hunk 10 above)
Comment (apgmartin) [inline css/article.css]: done

File: css/article.css (hunk 11 above)
Comment (apgmartin) [inline css/article.css]: done

File: css/article.css (hunk 12 above)
Comment (apgmartin) [inline css/article.css]: done

File: css/article.css (hunk 13 above)
Comment (apgmartin) [inline css/article.css]: done

File: css/article.css [hunk 29]
```diff
@@ -0,0 +1,204 @@
...
+
+#title_block h4 {
+    margin-top: 5px;
+    font-family: 'Satisfy', cursive;
+    text-align: right;
+    width: 300px;
+    font-size: 30px;
+    position: relative;
+    left: 605px;
+    color: white;
```
Comment (apgmartin) [inline css/article.css]: done

File: css/article.css (hunk 14 above)
Comment (apgmartin) [inline css/article.css]: done

File: css/article.css (hunk 15 above)
Comment (apgmartin) [inline css/article.css]: done

File: css/article.css (hunk 16 above)
Comment (apgmartin) [inline css/article.css]: done

File: css/article.css (hunk 17 above)
Comment (apgmartin) [inline css/article.css]: done

File: css/article.css (hunk 18 above)
Comment (apgmartin) [inline css/article.css]: done

File: css/blog.css (hunk 19 above)
Comment (apgmartin) [inline css/blog.css]: done

File: css/blog.css (hunk 20 above)
Comment (apgmartin) [inline css/blog.css]: done

File: css/blog.css (hunk 21 above)
Comment (apgmartin) [inline css/blog.css]: done

File: css/blog.css (hunk 22 above)
Comment (apgmartin) [inline css/blog.css]: done

File: css/blog.css (hunk 23 above)
Comment (apgmartin) [inline css/blog.css]: done

File: css/index.css [hunk 30]
```diff
@@ -71,12 +71,66 @@ a:hover .pc {
...
+    font-size: 22px;
+    margin-top: 0;
+    color: #FF6700;
+    text-decoration: underline;
+    text-align: right;
+}
+
+.blog_post h4 {
+    font-size: 12px;
+    color: white;
```
Comment (apgmartin) [inline css/index.css]: done

File: css/index.css (hunk 24 above)
Comment (apgmartin) [inline css/index.css]: done

File: index.php (hunk 25 above)
Comment (apgmartin) [inline index.php]: done

File: latest_blogs.php (hunk 26 above)
Comment (apgmartin) [inline latest_blogs.php]: done

File: tests/blog_class.php (hunk 27 above)
Comment (apgmartin) [inline tests/blog_class.php]: done

File: tests/latest_blogs.php [hunk 31]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+require_once ('../blog_class.php');
```
Comment (apgmartin) [inline tests/latest_blogs.php]: done

File: tests/random_class.php (hunk 28 above)
Comment (apgmartin) [inline tests/random_class.php]: done

**Pull request merged**